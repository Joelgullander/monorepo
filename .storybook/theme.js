import { create } from '@storybook/theming';

const theme = create({
    base: 'light',
  
    colorPrimary: '#D52B1E',
    colorSecondary: '#D52B1E',
  
    // UI
    appBg: 'white',
    appContentBg: '#f9f9f9',
    appBorderColor: '#f000',
    appBorderRadius: 0,
  
    // Typography
    fontBase: '"Roboto", sans-serif',
    fontCode: 'monospace',
  
    // Text colors
    textColor: '#212121',
    textInverseColor: 'rgba(255,255,255,0.9)',
  
    // Toolbar default and active colors
    barTextColor: 'white',
    barSelectedColor: '#D52B1E',
    barBg: '#000',
  
    // Form colors
    inputBg: 'white',
    inputBorder: 'silver',
    inputTextColor: 'black',
    inputBorderRadius: 4,
  
    brandTitle: 'Hiab components',
    brandUrl: 'https://hiab.com',
    brandImage: 'https://www.hiab.com/Content/images/logotypes/hiab2x2.png',
  });

  export default theme;