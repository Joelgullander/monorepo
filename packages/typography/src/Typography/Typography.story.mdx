import { Meta, Story, Props, Preview } from '@storybook/addon-docs/blocks';
import { withKnobs, text, boolean, number } from "@storybook/addon-knobs";
import { action } from '@storybook/addon-actions';

import { Typography, Test } from './Typography';

<Meta title='Components|Typography' component={Typography} decorators={[withKnobs]} />

# Typography

### Tagline

<Preview>
    <Story name="Tagline">
        <Typography type="Tagline">
            Products
        </Typography>
    </Story>
    <Story name="Tagline blue">
        <Typography type="Tagline" color="blue">
            Products
        </Typography>
    </Story>
</Preview>

### h1

<Preview>
     <Story name="h1">
        <Typography type="h1">
            LARGE TITLE Headline can flowovÅer mult iple lines of text så här fungerar ÅÄÖ
        </Typography>
    </Story>
</Preview>

### Meta

<Preview>
    <Story name="Meta">
        <Typography type="Meta">
            24 december 2019 — By Rayn Anderson — Product news
        </Typography>
    </Story>
</Preview>

### Ingress

<Preview>
     <Story name="Ingress">
        <Typography type="Ingress">
            In late ipsum dolor sit amet, consectetur adipiscing elit.
            Nam laoreet augue augue, in blandit ex rutrum sit Sno Aenean
            varius nulla nec justo mattis, venenatis tempus elit interdum.
            Donec sit amet iaculis nisl. Donec semper accumsan neque,
            nec dignissim dolor ultricies vitae.
        </Typography>
    </Story>
</Preview>

### Body

<Preview>
     <Story name="Body">
        <Typography style={{color: 'blue !important'}} type="Body">
            In late ipsum dolor sit amet, consectetur adipiscing elit.
            Nam laoreet augue augue, in blandit ex rutrum sit Sno Aenean varius nulla
            nec justo mattis, venenatis tempus elit interdum. Donec sit amet iaculis nisl
            Donec semper accumsan neque, nec dignissim dolor ultricies vitae.
            Aliquam sit amet velit molestie, lobortis nulla quis, solliudin mi.
            Donec semper accumsan neque, nec dignissim dolor ultricies vitae.
            Aliquam sit amet velit molestie, lobortis nulla quis, solliudin mi.
        </Typography>
    </Story>
</Preview>

### h2

<Preview>
     <Story name="h2">
        <Typography type="h2">
            Aliquam sit amet velit
        </Typography>
    </Story>
</Preview>

### h3

<Preview>
     <Story name="h3">
        <Typography type="h3">
            Aliquam sit amet velit
        </Typography>
    </Story>
</Preview>

### h4

<Preview>
     <Story name="h4">
        <Typography type="h4">
            Aliquam sit amet velit
        </Typography>
    </Story>
</Preview>

### h5

<Preview>
     <Story name="h5">
        <Typography type="h5">
            Aliquam sit amet velit
        </Typography>
    </Story>
</Preview>

### EmfasItalic

<Preview>
     <Story name="EmfasItalic">
        <Typography type="EmfasItalic">
            Aliquam sit amet velit
        </Typography>
    </Story>
</Preview>

### Line

<Preview>
     <Story name="Line">
        <Typography type="Line" />
    </Story>
</Preview>

### Blockquote

<Preview>
     <Story name="Blockquote">
        <Typography type="Blockquote">
            In latenean variusius nulla nec justo mattis, venenatis tempus elit interdum. Donec sit amet iaculis nisl. Donec semper accumsan neque, nec dignissim dolor ultricies vitae. Aliquam sit amet velit molestie, lobortis nulla quis, sollicitudin mi.
        </Typography>
    </Story>
</Preview>

### List bullet

<Preview>
     <Story name="List Bullet">
        <Typography type="List">
            <Typography type="ListItem">Donec semper accumsan neque, nec dignissim dolor ultricies vitae. Aliquam sit amet velit molestie, lobortis nulla quis, sollicitudin mi.</Typography>
            <Typography type="ListItem">Donec semper accumsan neque, nec dignissim dolor ultricies vitae. Aliquam sit amet velit molestie, lobortis nulla quis, sollicitudin mi.</Typography>
            <Typography type="ListItem">Donec semper accumsan neque, nec dignissim dolor ultricies vitae. Aliquam sit amet velit molestie, lobortis nulla quis, sollicitudin mi.</Typography>
        </Typography>
    </Story>
</Preview>

### List ordered

<Preview>
     <Story name="List Ordered">
        <Typography listType="ordered" type="List">
            <Typography type="ListItem">Donec semper accumsan neque, nec dignissim dolor ultricies vitae. Aliquam sit amet velit molestie, lobortis nulla quis, sollicitudin mi.</Typography>
            <Typography type="ListItem">Donec semper accumsan neque, nec dignissim dolor ultricies vitae. Aliquam sit amet velit molestie, lobortis nulla quis, sollicitudin mi.</Typography>
            <Typography type="ListItem">Donec semper accumsan neque, nec dignissim dolor ultricies vitae. Aliquam sit amet velit molestie, lobortis nulla quis, sollicitudin mi.</Typography>
        </Typography>
    </Story>
</Preview>

### Big callout

<Preview>
    <Story name="Big callout">
        <Typography type="Callout">
            In late ipsum dolor sit amet, consectetur adipiscing elit. Nam laoreet augue augue, in blandit ex rutrum sits nulla nec justo mattis, venenatis tempus elit interdum. Donec sit amet iaculis nisl. Donec semper accumsan neque, nec dignissim dolor ultricies vitae. Aliquam sit amet velit molestie, lobortis nulla quis, sollicitudin mi. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam laoreet augue augue, in blandit ex rutrum sit amet. Aenean varius nulla nec justo mattis, venenatis tempus elit interdum. Donec sit amet iaculis nisl. Donec semper accumsan neque, nec dignissim dolor ultricies vitae. Aliquam sit amet velit molestie, lobortis nulla quis, sollicitudin mi.
        </Typography>
    </Story>
</Preview>

## Display Types

### Large

<Preview>
    <Story name="h1 display">
        <Typography type="h1Display">
            world class equipment
        </Typography>
    </Story>
</Preview>

### Medium

<Preview>
    <Story name="h2 display">
        <Typography type="h2Display">
            world class equipment
        </Typography>
    </Story>
</Preview>

### Medium Light

<Preview>
    <Story name="h2 display light">
        <Typography type="h2DisplayLight">
            world class equipment
        </Typography>
    </Story>
</Preview>

### Small

<Preview>
    <Story name="h3 display">
        <Typography type="h3Display">
            world class equipment
        </Typography>
    </Story>
</Preview>

### Mini

<Preview>
    <Story name="h4 display">
        <Typography type="h4Display">
            world class equipment
        </Typography>
    </Story>
</Preview>

## Props

<Props of={Typography} />
