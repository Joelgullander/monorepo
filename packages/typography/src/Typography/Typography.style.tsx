import styled from 'styled-components';
import { HiabTheme } from '@hiab/theme';
import { ListType } from './Typography';

interface IStyledTypography {
    theme: HiabTheme,
    listType?: ListType,
    color?: string,
};

export const Tagline = styled.div`
    color: ${(props: IStyledTypography) => props.color || props.theme.colors.thunderbird};
    font: 17px ${(props: IStyledTypography) => props.theme.fonts.primary};
    font-weight: 700;
    text-transform: uppercase;
    letter-spacing: 1.7px;
    margin: 0 0 30px 0;
`;

export const h1 = styled.h1`
    color: ${(props: IStyledTypography) => props.color || props.theme.colors.mineshaft};
    font: 47px/60px ${(props: IStyledTypography) => props.theme.fonts.secondary};
    text-transform: uppercase;
    letter-spacing: 1.88px;
    margin: 0 0 30px 0;
`;

export const Ingress = styled.div`
    color: ${(props: IStyledTypography) => props.color || props.theme.colors.mako};
    font: 22px/33px ${(props: IStyledTypography) => props.theme.fonts.primary};
    letter-spacing: 0px;
    font-weight: 300;
    margin: 0 0 60px 0;
`;

export const h2 = styled.h2`
    color: ${(props: IStyledTypography) => props.color || props.theme.colors.mineshaft};
    font: 34px/43px ${(props: IStyledTypography) => props.theme.fonts.primary};
    font-weight: 700;
    margin: 0 0 20px 0;
`;

export const h3 = styled.h3`
    color: ${(props: IStyledTypography) => props.color || props.theme.colors.mineshaft};
    font: 24px/31px ${(props: IStyledTypography) => props.theme.fonts.primary};
    font-weight: 700;
    margin: 0 0 20px 0;
`;

export const h4 = styled.h4`
    color: ${(props: IStyledTypography) => props.color || props.theme.colors.mineshaft};
    font: 20px/26px ${(props: IStyledTypography) => props.theme.fonts.primary};
    font-weight: 700;
    margin: 0 0 20px 0;
`;

export const h5 = styled.h5`
    color: ${(props: IStyledTypography) => props.color || props.theme.colors.mineshaft};
    font: 18px/24px ${(props: IStyledTypography) => props.theme.fonts.primary};
    font-weight: 400;
    margin: 0 0 20px 0;
`;

export const Meta = styled.div`
    color: ${(props: IStyledTypography) => props.color || props.theme.colors.osloGray};
    font: 17px ${(props: IStyledTypography) => props.theme.fonts.primary};
    margin: 0 0 50px 0;
`;

export const Body = styled.p`
    color: ${(props: IStyledTypography) => props.color || props.theme.colors.mako};
    font: 16px/25px ${(props: IStyledTypography) => props.theme.fonts.primary};
    margin: 0 0 50px 0;
`;

export const EmfasItalic = styled.div`
    color: ${(props: IStyledTypography) => props.color || props.theme.colors.mako};
    font: italic 22px/33px ${(props: IStyledTypography) => props.theme.fonts.primary};
    margin: 0 0 40px 0;
`;

export const Blockquote = styled.blockquote`
    color: ${(props: IStyledTypography) => props.color || props.theme.colors.mako};
    font: 15px/25px ${(props: IStyledTypography) => props.theme.fonts.primary};
    border-left: 5px solid ${(props: IStyledTypography) => props.theme.colors.thunderbird};
    padding: 10px 0 10px 20px;
    font-weight: 700;
    margin: 0 0 20px 0;
`;

export const Line = styled.hr`
    border: 0;
    background: ${(props: IStyledTypography) => props.color || '#D6D6D6'};
    height: 2px;
`;

export const List = styled.ul`
    color: ${(props: IStyledTypography) => props.color || props.theme.colors.mako};
    font: 16px/25px ${(props: IStyledTypography) => props.theme.fonts.primary};
    margin: 0 0 50px 0;
    ${(props: IStyledTypography) => {
        if (props.listType === 'ordered')
            return `
                list-style: none;
                counter-reset: li;

                & > li {
                    counter-increment: li;

                    &:before {
                        content: counter(li);
                        color: ${props.theme.colors.black};
                        display: inline-block;
                        margin-left: -35px;
                        line-height: 10px;
                        font-size: 17px;
                        position: absolute;
                        top: 14px;
                        font-weight: 900;
                    }  
                }
            `;
        if (props.listType === 'bullet')
            return `
                list-style: none;
               
                & > li {
                    &:before {
                        content: "â€¢";
                        color: ${props.theme.colors.thunderbird};
                        display: inline-block;
                        width: 28px;
                        margin-left: -35px;
                        line-height: 10px;
                        font-size: 30px;
                        position: absolute;
                        top: 14px;
                    }  
                }
            `;
        return 'list-style: none';
    }}
`;

export const ListItem = styled.li`
    padding: 5px 0 5px 18px;
    position: relative;  
`;

export const Callout = styled.div`
    font: 22px/33px ${(props: IStyledTypography) => props.theme.fonts.primary};
    font-weight: 300;
    color: ${(props: IStyledTypography) => props.color || props.theme.colors.osloGray};
    margin: 0 0 60px 0;
`;

export const h1Display = styled.h1`
    color: ${(props: IStyledTypography) => props.color || props.theme.colors.black};
    font: 47px/60px ${(props: IStyledTypography) => props.theme.fonts.secondary};
    text-transform: uppercase;
    letter-spacing: 1.88px;
    margin: 0 0 30px 0;
`;

export const h2Display = styled.h2`
    color: ${(props: IStyledTypography) => props.color || props.theme.colors.black};
    font: 32px/44px ${(props: IStyledTypography) => props.theme.fonts.secondary};
    text-transform: uppercase;
    letter-spacing: 0;
    margin: 0 0 30px 0;
`;

export const h2DisplayLight = styled.h2`
    color: ${(props: IStyledTypography) => props.color || props.theme.colors.mineshaft};
    font: 34px/43px ${(props: IStyledTypography) => props.theme.fonts.primary};
    letter-spacing: 1.88px;
    margin: 0 0 30px 0;
`;

export const h3Display = styled.h3`
    color: ${(props: IStyledTypography) => props.color || props.theme.colors.black};
    font: 28px/38px ${(props: IStyledTypography) => props.theme.fonts.secondary};
    text-transform: uppercase;
    letter-spacing: 1.12px;
    margin: 0 0 30px 0;
`;

export const h4Display = styled.h4`
    color: ${(props: IStyledTypography) => props.color || props.theme.colors.black};
    font: 18px/26px ${(props: IStyledTypography) => props.theme.fonts.secondary};
    text-transform: uppercase;
    letter-spacing: 0.72px;
    margin: 0 0 30px 0;
`;
