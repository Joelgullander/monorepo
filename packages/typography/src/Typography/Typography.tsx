import React from 'react';
import * as styled from './Typography.style';

export enum Types {
  TAGLINE = 'Tagline',
  H1 = 'h1',
  META = 'Meta',
  INGRESS = 'Ingress',
  H2 = 'h2',
  H3 = 'h3',
  H4 = 'h4',
  H5 = 'h5',
  BODY = 'Body',
  EMFAS_ITALIC = 'EmfasItalic',
  LINE = 'Line',
  LIST = 'List',
  LIST_ITEM = 'ListItem',
  BLOCKQUOTE = 'Blockquote',
  CALLOUT = 'Callout',
  H1_DISPLAY = 'h1Display',
  H2_DISPLAY = 'h2Display',
  H2_DISPLAY_LIGHT = 'h2DisplayLight',
  H3_DISPLAY = 'h3Display',
  H4_DISPLAY = 'h4Display',
};

export type ListType = 'bullet' | 'ordered';

export interface ITypography {
  color?: string,
  styles?: React.CSSProperties,
  listType?: ListType,
  type: Types,
};

export const Typography: React.FC<ITypography> = (props) => {
  switch (props.type) {
    case Types.TAGLINE:
      return (
        <styled.Tagline color={props.color} style={props.styles}>
          {props.children}
        </styled.Tagline>
      );
    case Types.H1:
      return (
        <styled.h1 color={props.color} style={props.styles}>
          {props.children}
        </styled.h1>
      );
    case Types.H2:
      return (
        <styled.h2 color={props.color} style={props.styles}>
          {props.children}
        </styled.h2>
      );
    case Types.H3:
      return (
        <styled.h3 color={props.color} style={props.styles}>
          {props.children}
        </styled.h3>
      );
    case Types.H4:
      return (
        <styled.h4 color={props.color} style={props.styles}>
          {props.children}
        </styled.h4>
      );
    case Types.H5:
      return (
        <styled.h5 color={props.color} style={props.styles}>
          {props.children}
        </styled.h5>
      );
    case Types.META:
      return (
        <styled.Meta color={props.color} style={props.styles}>
          {props.children}
        </styled.Meta>
      );
    case Types.INGRESS:
      return (
        <styled.Ingress color={props.color} style={props.styles}>
          {props.children}
        </styled.Ingress>
      );
    case Types.BODY:
      return (
        <styled.Body color={props.color} style={props.styles}>
          {props.children}
        </styled.Body>
      );
    case Types.EMFAS_ITALIC:
      return (
        <styled.EmfasItalic color={props.color} style={props.styles}>
          {props.children}
        </styled.EmfasItalic>
      );
    case Types.LINE:
      return (
        <styled.Line color={props.color} style={props.styles} />
      );
    case Types.BLOCKQUOTE:
      return (
        <styled.Blockquote color={props.color} style={props.styles} >
          {props.children}
        </styled.Blockquote>
      );
    case Types.LIST:
      return (
        <styled.List
          listType={props.listType}
          style={props.styles}
          color={props.color}
        >
          {props.children}
        </styled.List>
      );
    case Types.LIST_ITEM:
      return (
        <styled.ListItem style={props.styles}>
          {props.children}
        </styled.ListItem>
      );
    case Types.CALLOUT:
      return (
        <styled.Callout color={props.color} style={props.styles}>
          {props.children}
        </styled.Callout>
      );
    case Types.H1_DISPLAY:
      return (
        <styled.h1Display color={props.color} style={props.styles}>
          {props.children}
        </styled.h1Display>
      );
    case Types.H2_DISPLAY:
      return (
        <styled.h2Display color={props.color} style={props.styles}>
          {props.children}
        </styled.h2Display>
      );
    case Types.H2_DISPLAY_LIGHT:
      return (
        <styled.h2DisplayLight color={props.color} style={props.styles}>
          {props.children}
        </styled.h2DisplayLight>
      );
    case Types.H3_DISPLAY:
      return (
        <styled.h3Display color={props.color} style={props.styles}>
          {props.children}
        </styled.h3Display>
      );
    case Types.H4_DISPLAY:
      return (
        <styled.h4Display color={props.color} style={props.styles}>
          {props.children}
        </styled.h4Display>
      );
    default:
      return null;
  }
};

Typography.defaultProps = {
  type: Types.TAGLINE,
  listType: 'bullet',
};
