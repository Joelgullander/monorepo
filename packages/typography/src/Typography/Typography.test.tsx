import * as React from 'react';

import { shallow, mount } from 'enzyme';
import { Typography, Types } from './Typography';
import { ThemeProvider } from '@hiab/theme';

const ThemeProviderWrapper : React.FC = ({ children }) => (
  <ThemeProvider>
    { children }
  </ThemeProvider>
);

export const shallowWithTheme = (tree: React.ReactElement) => shallow(tree, {
  wrappingComponent: ThemeProviderWrapper
});

export const mountWithTheme = (tree: React.ReactElement) => mount(tree, {
  wrappingComponent: ThemeProviderWrapper
});


describe('<Typography />', () => {
  test('renders without exploding', () => {
    const wrapper = shallowWithTheme(<Typography type={Types.H1} />)
    expect(wrapper.exists()).toBe(true);
  });

  test('displays text based on props.children', () => {
    const tpgr = shallow(<Typography type={Types.H1}>Header H1</Typography>);

    expect(tpgr.text()).toBe('Header H1');

    tpgr.setProps({ children: 'Header H1 UPD' });

    expect(tpgr.text()).toBe('Header H1 UPD')
  });

  test('snapshot', () => {
    const wrapper = shallowWithTheme(<Typography type={Types.H1}>Typography</Typography>);

    expect(wrapper).toMatchSnapshot();
  });

  test('prop type is passing', () => {
    const wrapper = mountWithTheme(
        <Typography type={Types.H1}>Typography</Typography>
    );
    
    expect(wrapper.props().type).toEqual(Types.H1);
  });

  test('prop color is passing', () => {
    const wrapper = mountWithTheme(
        <Typography color="#fff" type={Types.H1}>Typography</Typography>
    );
    
    expect(wrapper.props().color).toEqual("#fff");
  });
});
