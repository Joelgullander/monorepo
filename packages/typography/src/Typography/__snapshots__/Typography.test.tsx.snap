// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Typography /> snapshot 1`] = `
<styled.h1>
  Typography
</styled.h1>
`;
